### PR TITLE
Added an integration test to test receiving sample upload finished messages

### PR DESCRIPTION
**Pull request by madeye-matt**
Pattern based on SampleEndpointIT.java from the sample service.

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            List<SampleLinkDTO> links = this.client.getSampleLinks(newCollex.getId());
+
+            assertEquals(1, links.size());
+
+            SampleLinkDTO link = links.get(0);
+
+            assertEquals(sampleSummaryId, UUID.fromString(link.getSampleSummaryId()));
+            assertEquals("ACTIVE", link.getState());
+
+        } catch (JsonProcessingException e) {
```
Comment (benjefferies) [inline src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java]: Not a bigger but seems weird to try catch in the test. Why not just throw Exception on the test and we'll get the stacktrace

File: src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            SampleSummaryDTO sampleSummary = new SampleSummaryDTO();
+            sampleSummary.setId(sampleSummaryId);
+
+            // This will cause an exception to be thrown as there is no collection instrument service but this is
+            // harmless to our purpose
+            sender.sendMessage("sample-outbound-exchange",
+                    "Sample.SampleUploadFinished.binding",
+                    this.mapper.writeValueAsString(sampleSummary));
+
+            Thread.sleep(500);
```
Comment (benjefferies) [inline src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java]: Seems sketchy to have a Thread.sleep here. Let's change it to some kind of backing off retrying policy. https://github.com/rholder/guava-retrying is good for that sort of thing

File: src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseClient.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        this.password = aPassword;
+
+        initialiseUnirestObjectMapper();
+    }
+
+    /**
+     * Initialises object mapper as used by unirest (needs a Jackson ObjectMapper to construct)
+     */
+    private void initialiseUnirestObjectMapper() {
+        Unirest.setObjectMapper(new com.mashape.unirest.http.ObjectMapper() {
```
[the remark this answers is not in the document]
Comment (madeye-matt) [reply src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseClient.java]: This class uses 2 different ObjectMapper classes: Unirest uses a unirest packaged ObjectMapper which is constructed from a normal Jackson one.  It's ugly but it's a "feature" of the unirest library.

File: src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java (hunk 1 above)
Comment (madeye-matt) [reply src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java]: Yes, will change - not sure why I did it like that, looks odd to me too

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpoint.java [hunk 4]
```diff
@@ -460,13 +465,30 @@ private void validateConstraints(final CollectionExerciseDTO collexDto) throws C
...
+     * @param collectionExerciseId the collection exercise for which the links are required
+     * @return a list of sample summaries linked to the collection exercise
+     */
+    @RequestMapping(value = "/link/{collectionExerciseId}", method = RequestMethod.GET,
+            produces = "application/vnd.ons.sdc.samplelink.v1+json")
+    public ResponseEntity<List<SampleLinkDTO>> getSampleLinks(
+            @PathVariable("collectionExerciseId") final UUID collectionExerciseId) {
+        log.debug("Getting linked sample summaries for {}", collectionExerciseId);
+        List<SampleLink> sampleLinks = this.collectionExerciseService.findLinkedSampleSummaries(collectionExerciseId);
+        List<SampleLinkDTO> sampleLinkDtoList = mapperFacade.mapAsList(sampleLinks, SampleLinkDTO.class);
```
[the remark this answers is not in the document]
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpoint.java]: I have a slight preference for Dto for aesthetic reasons but I think we should use whatever is already most prevalent in the code - I haven't checked so any idea which?  If not I'll do a quick straw poll of the Java modules

[the remark this answers is not in the document]
Comment (madeye-matt) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpoint.java]: Looks like there's a vast majority of class names with DTO but vars are Dto (but that's mostly my code :-P ).  Tell you what: I'll change it to sampleLinkList

**Pull request merged**